platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service;
+
+import com.hedera.hapi.node.state.primitives.ProtoBytes;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterState;
+import com.hedera.hapi.node.state.roster.RosterState.Builder;
+import com.hedera.hapi.node.state.roster.RoundRosterPair;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.platform.roster.RosterUtils;
+import com.swirlds.state.spi.ReadableStates;
+import com.swirlds.state.spi.WritableKVState;
+import com.swirlds.state.spi.WritableSingletonState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Provides write methods for interacting with Rosters.
+ */
+public class WritableRosterStore extends ReadableRosterStoreImpl {
+
+    public WritableRosterStore(@NonNull final ReadableStates states) {
+        super(states);
+    }
+
+    @Override
+    protected WritableKVState<ProtoBytes, Roster> rosters() {
+        return super.rosters();
+    }
+
+    @Override
+    protected WritableSingletonState<RosterState> rosterState() {
+        return super.rosterState();
+    }
+
+    /**
+     * Sets the active roster.
+     *
+     * @param roster the roster to set
+     *  @param round the round in which this roster became active
+     */
+    public void setActiveRoster(final Roster roster, final long round) {
+        Objects.requireNonNull(roster);
+        final RosterState previousRosterState = rosterState().get();
+        Objects.requireNonNull(previousRosterState);
```

Comment:
Is this always non-null?   What about the first time the roster is set?

Reply:
yes it's always non-null as a default is initialized in the schema (`V0540RosterSchema`).

Reply:
But I've now renamed it 'existing' as against 'previous' to make it clearer.